src/libutil/experimental-features.cc

```diff
@@ -5,29 +5,131 @@
 
 namespace nix {
 
-std::map<ExperimentalFeature, std::string> stringifiedXpFeatures = {
-    { Xp::CaDerivations, "ca-derivations" },
-    { Xp::ImpureDerivations, "impure-derivations" },
-    { Xp::Flakes, "flakes" },
-    { Xp::NixCommand, "nix-command" },
-    { Xp::RecursiveNix, "recursive-nix" },
-    { Xp::NoUrlLiterals, "no-url-literals" },
-    { Xp::FetchClosure, "fetch-closure" },
-    { Xp::ReplFlake, "repl-flake" },
-    { Xp::AutoAllocateUids, "auto-allocate-uids" },
-    { Xp::Cgroups, "cgroups" },
-    { Xp::DiscardReferences, "discard-references" },
+struct ExperimentalFeatureDetails
+{
+    ExperimentalFeature tag;
+    std::string_view name;
+    std::string_view description;
 };
 
+constexpr std::array<ExperimentalFeatureDetails, 11> xpFeatureDetails = {{
+    {
+        .tag = Xp::CaDerivations,
+        .name = "ca-derivations",
+        .description = R"(
+            Allows derivations to be content-addressed in order to prevent rebuilds
+            when changes to the derivation do not result in changes to the
+            derivation's output. See
+            [__contentAddressed](../language/advanced-attributes.md#adv-attr-__contentAddressed)
```

Comment:
```suggestion
            [__contentAddressed](@docroot@/language/advanced-attributes.md#adv-attr-__contentAddressed)
```

Use `@docroot@` in all links, then we don't have to keep in mind where the generated files end up.